lib/src/common/polymer_descriptor.dart

```diff
@@ -202,7 +202,13 @@ JsObject _jsType(Type type) {
       return context['String'];
     case 'Map':
     case 'JsObject':
-    default:
       return context['Object'];
+    default:
+      // Store the type in the same manner as done in _addDartInstance so the
+      // type is retrieved in dartValue through _getDartInstance.
+      var value = new JsObject(context['Object']);
```

Comment:
When I attempt to just return the value here I end up with the following exception

`TypeError: Cannot redefine property: __dartClass__`

Looks like this has to do with the propertySetterFactory.